lib/solidus_support/engine_extensions.rb

```diff
@@ -82,17 +82,23 @@ def solidus_subscribers_root
       #
       # @see #load_solidus_decorators_from
       def enable_solidus_engine_support(engine)
-        paths['app/controllers'] << "lib/controllers/#{engine}"
-        paths['app/views'] << "lib/views/#{engine}"
+        initializer "#{name}_#{engine}_paths", before: :initialize_dependency_mechanism do
+          if SolidusSupport.send(:"#{engine}_available?")
```

Comment:
Does this `if` need to be here (as opposed to wrapping the whole method) because `SolidusSupport.send(:"#{engine}_available?")` needs to be evaluated when the initializer block is executed? If yes, maybe it's worth documenting it, at least in the commit message.

Reply:
That's it. Outside the `initializer` block, we're still at load time execution time, so the frontend might not be available yet. I updated the commit message to make it clearer.